addon_imps/storage/gitlab.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+import urllib
+from dataclasses import dataclass
+from http import HTTPStatus
+from urllib.parse import quote_plus
+
+from addon_imps.storage.utils import ItemResultable
+from addon_service.common.exceptions import ItemNotFound
+from addon_toolkit.interfaces import storage
+from addon_toolkit.interfaces.storage import (
+    ItemResult,
+    ItemSampleResult,
+    ItemType,
+)
+
+
+FOLDER_ITEM_TYPES = frozenset(["subfolder", "tree", "folder"])
+
+
+class GitlabStorageImp(storage.StorageAddonHttpRequestorImp):
+    """storage on gitlab
+
+    see https://developers.google.com/drive/api/reference/rest/v3/
+    """
+
+    async def get_external_account_id(self, _: dict[str, str]) -> str:
+        async with self.network.GET("user/preferences") as response:
+            resp_json = await response.json_content()
+            return resp_json.get("user_id", "")
```

Comment:
I'm not sure but isn't the **Gitlab** api returns `users id` using `self.network.GET("user")` and than just get `"id"` from response instead of `user/preferences` ? 
So it would look something like this:
```
    async with self.network.GET("user") as response:
        resp_json = await response.json_content()
        return str(resp_json["id"])
```

Reply:
This endpoint is `user/preferences`
I used this enpoint because it returns smaller json)
https://docs.gitlab.com/ee/api/users.html#get-your-user-preferences

Reply:
Gotcha, seems reasonable) Leave it as it is then